src/print_utils.h

```diff
@@ -13,7 +13,7 @@
 
 /* Hexadecimal output utils */
 
-static const char hex_chars[16] = "0123456789abcdef";
+static const char hex_chars[17] = "0123456789abcdef";
```

Comment:
Is there any way to tell the compiler that `hex_chars` is *not* a C string?
If not, then I'd rather have an ugly `{ '0', '1', ..., 'f' }` initialization instead.
In any case, let's add a comment next to the initialization explaining the source of this ugliness.

Reply:
AFAIU there is no nice way to annotate it. Switched to `{ '0', '1', ..., 'f' }` array style initializer and added a comment.